src/object_store.cpp

```diff
@@ -529,7 +527,7 @@ static void apply_pre_migration_changes(Group& group, std::vector<SchemaChange>
         void operator()(ChangePropertyType op) { replace_column(group, table(op.object), *op.old_property, *op.new_property); }
         void operator()(MakePropertyNullable op) { make_property_optional(table(op.object), *op.property); }
         void operator()(MakePropertyRequired op) { make_property_required(group, table(op.object), *op.property); }
-        void operator()(ChangePrimaryKey op) { ObjectStore::set_primary_key_for_object(group, op.object->name.c_str(), op.property ? op.property->name.c_str() : ""); }
+        void operator()(ChangePrimaryKey op) { table(op.object).set_primary_key_column(ColKey{}); }
```

Comment:
When we're changing the primary key we want to remove the old one at the start of the migration and then add the new one at the end because the user may be populating the new PK during the migration.